zimui/src/stores/main.ts

```diff
@@ -1,52 +1,89 @@
 import { defineStore } from 'pinia'
-import axios from 'axios'
+import axios, { AxiosError }from 'axios'
 import Channel from '@/types/Channel'
 import Topic from '@/types/Topic'
 
 export type RootState = {
   channelData: Channel | null
   isLoading: boolean
   errorMessage: string
+  error: AxiosError<object> | null
 }
 export const useMainStore = defineStore('main', {
   state: () =>
     ({
       channelData: null,
       isLoading: false,
       errorMessage: '',
+      error: null,
     }) as RootState,
   getters: {},
   actions: {
     async fetchChannel() {
       this.isLoading = true
       this.errorMessage = ''
-      return axios.get('./channel.json').then(
-        (response) => {
-          this.isLoading = false
-          this.channelData = response.data as Channel
-        },
-        (_) => {
+      try {
+        const response = await axios.get('./channel.json')
+        this.isLoading = false
+        this.channelData = response.data as Channel
+      } catch (error) {
+        if (error instanceof AxiosError) {
           this.isLoading = false
```

Comment:
You must set `this.isLoading = false` and `this.channelData = null` in all cases, not just if error is AxiosError